Training_25/Program.cs

```diff
@@ -3,11 +3,64 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program on T01 branch - Assignments Q1
 // ------------------------------------------------------------------------------------------------
+using System.Text;
 namespace Training_25;
+
 internal class Program {
    static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+      Console.WriteLine ("Enter a number");
+      string? input = Console.ReadLine ();
+      int num;
+      while (!int.TryParse (input, out num)) {
+         Console.WriteLine ("Invalid input.Please enter a valid number:");
+         input = Console.ReadLine ();
+      }

```

Comment:
```suggestion
      while (!int.TryParse (Console.ReadLine (), out num))
         Console.WriteLine ("Invalid input. Please enter a valid number:");
```
Since the input variable is used only in one place there is no need to cache (store) the value. You can directly used it for ReadLine (); Also, please add spaces after punctuation, it improves readability.

Reply:
<img width="830" height="384" alt="Image" src="https://github.com/user-attachments/assets/8ca1e4db-fd26-4333-9b03-cbadb9f1cb14" />
It's also helpful to mention what is the expected output when you say "invalid input".